setup.py

```diff
@@ -1,41 +1,58 @@
-from setuptools import setup, find_packages
+import sys
+
 import param
+from setuptools import find_packages, setup
 
 extras_require = {
     'tests': [
-        'pytest',
         'codecov',
-        'pytest-cov',
         'flake8',
+        'geopandas',
         'hypothesis',
+        'pytest-cov',
+        'pytest',
         'scipy',
         'shapely',
-        'geopandas',
     ],
     'examples': [
-        'geopandas',
-        'matplotlib',
-        'descartes',
         'datashader',
+        'descartes',
+        'geopandas',
         'holoviews',
+        'matplotlib',
     ]
 }
 
 install_requires = [
-    'pandas>=0.25',
     'dask[complete] >=2.0',
+    'fsspec',
     'numba',
-    'numpy',
-    'pyarrow>=0.15',
+    'pandas>=0.25',
     'param',
-    'fsspec',
     'retrying',
+    'snappy',
 ]
 
+# Checking for platform explicitly because
+# pyctdev does not handle dependency conditions
+# such as 'numpy<1.20;platform_system=="Darwin"'
+if sys.platform == 'darwin':
+    install_requires.extend([
+        'numpy<1.20',
+        'pyarrow==2',
+    ])
+else:
+    install_requires.extend([
+        'numpy',
+        'pyarrow>=0.15',
```

Comment:
Is there a reason for pyarrow to be specified in a platform-dependent way? It sounds like numpy has platform-specific issues, but I don't know of any with pyarrow, so I would think it should have just one pinning either >=0.15 or >=2, depending on whether we think the performance improvements are worth the loss in flexibility. I don't have much preference between those two, and don't know of anything that would force it to go one way or the other now that pyarrow=2 (and =3) on the default Anaconda channels (as of last week).

Reply:
I should have added a note that pyarrow was causing failures on Mac.  I tried a number of various combinations and the only way I was able to get it working on Mac was to pin below version 2.  This pin did not seem necessary on other platforms, which is why it was made platform  dependent.  I am testing installing pyarrow from conda-forge next based on the link you provided above.

Reply:
I was able to remove pyarrow from platform specific pins, but had to pin dask on Mac.